src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -0,0 +1,714 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Security;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Microsoft.Win32;
+
+using Internal.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// This class contains methods that are mainly used to marshal between unmanaged
+    /// and managed types.
+    /// </summary>
+    public static partial class Marshal
+    {
+        /// <summary>
+        /// The default character size for the system. This is always 2 because
+        /// the framework only runs on UTF-16 systems.
+        /// </summary>
+        public static readonly int SystemDefaultCharSize = 2;
+
+        /// <summary>
+        /// The max DBCS character size for the system.
+        /// </summary>
+        public static readonly int SystemMaxDBCSCharSize = GetSystemMaxDBCSCharSize();
+
+        public static unsafe string PtrToStringAnsi(IntPtr ptr)
+        {
+            if (IntPtr.Zero == ptr)
```

Comment:
Most of the checks in this file have `IntPtr.Zero` on the right side. This one and a few others have it on the left side. Unify? 

Also, the IntPtr.Zero checks can be together with the IsWin32Atom in a single if.

```
if ((ptr == IntPtr.Zero) || IsWin32Atom(ptr))
{
    return null;
}
```

Reply:
Nit: Drop the extra parenthesis and make it just `if (ptr == IntPtr.Zero || IsWin32Atom(ptr))`.